libs/sdk-core/src/breez_services.rs

```diff
@@ -1220,55 +1220,78 @@ impl BreezServices {
+            .await?;
+
+        let maybe_closed_at = maybe_outspend
+            .clone()
+            .and_then(|outspend| outspend.status)
+            .and_then(|s| s.block_time);
+        let maybe_closing_txid = maybe_outspend.and_then(|outspend| outspend.txid);
+
+        Ok((maybe_closed_at, maybe_closing_txid))
+    }
+
     async fn closed_channel_to_transaction(
         &self,
         channel: crate::models::Channel,
     ) -> Result<Payment> {
-        let now_epoch_sec = SystemTime::now().duration_since(UNIX_EPOCH)?.as_secs();
-
-        let channel_closed_at = match channel.closed_at {
-            Some(closed_at) => closed_at,
-            None => {
-                // If we don't have it, we look it up from the channel closing tx
-                let looked_up_channel_closed_at = match channel.funding_outnum {
+        let (payment_time, closing_txid) = match (channel.closed_at, channel.closing_txid.clone()) {
+            (Some(closed_at), Some(closing_txid)) => (closed_at as i64, closing_txid),
+            (_, _) => {
+                // If any of the two closing-related fields are empty, we look them up and persist them
+                let (maybe_closed_at, maybe_closing_txid) =
+                    self.lookup_channel_closing_data(&channel).await?;
+
+                let processed_closed_at = match maybe_closed_at {
                     None => {
-                        warn!("No founding_outnum found for the closing tx, defaulting closed_at to epoch time");
-                        now_epoch_sec
-                    }
-                    Some(outnum) => {
-                        // Find the output tx that was used to fund the channel
-                        let outspends = self
-                            .chain_service
-                            .transaction_outspends(channel.funding_txid.clone())
-                            .await?;
-                        let maybe_block_time = outspends
-                            .get(outnum as usize)
-                            .and_then(|outspend| outspend.status.as_ref())
-                            .and_then(|status| status.block_time);
-
-                        match maybe_block_time {
-                            None => {
-                                warn!("Blocktime could not be determined for funding_outnum {outnum}, defaulting closed_at to epoch time");
-                                now_epoch_sec
-                            }
-                            Some(block_time) => block_time,
-                        }
+                        warn!("Blocktime could not be determined for from closing outspend, defaulting closed_at to epoch time");
+                        SystemTime::now().duration_since(UNIX_EPOCH)?.as_secs()
                     }
+                    Some(block_time) => block_time,
                 };
+                let processed_closing_txid = maybe_closing_txid.unwrap_or_default();
```

Comment:
It looks like we persisting defaults for both closed_at and the closing_tx_id in case we didn't find the outspend.
It looks like it prevents us from fetching it next time.

Reply:
Yes, I thought that's desired behavior. If outspend not found:
- `closed_at` is saved as `now()`
- `closing_txid` is saved as empty string

If we don't persist them in this case, yes we will try to look it up again next time, but until then the fields would be empty. Is that how it should actually be?